server/src/main/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -630,6 +682,21 @@ private Decision earlyTerminate(RoutingAllocation allocation, final Map<String,
         return null;
     }
 
+    /**
+     * Determine the shards that satisfy the predicate
+     * @param predicate shard routing predicate which needs to be met
+     * @return List of shards
+     */
+    List<ShardRouting> shardsSatisfyingPredicate(RoutingNode routingNode, Predicate<ShardRouting> predicate) {
```

Comment:
I'd probably remove this method and just do the following inline:

```
StreamSupport.stream(node.spliterator(), false)
    .filter(<predicate>)
```

It looks like you only ever use the resulting list as a stream in non-test code, so inlining as a stream will avoid the intermediate list all together. If something like this is helpful in a test then you can add a helper method in the test class.

Reply:
Sure. I can move it inline for now. 
I think it stemmed out of being a useful alternative within `ShardRouting` but does not seem as useful right away within `DiskThresholdDecider`.